app/web/features/auth/notifications/NotificationSettings.tsx

```diff
@@ -0,0 +1,87 @@
+import { Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import CircularProgress from "components/CircularProgress";
+import { notificationSettingsQueryKey } from "features/queryKeys";
+import { Error as GrpcError } from "grpc-web";
+import {
+  GetNotificationSettingsRes,
+  SetNotificationSettingsRes,
+} from "proto/notifications_pb";
+import { useTranslation } from "react-i18next";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { service } from "service";
+
+interface NotificationSettingFormData {
+  newNotificationsEnabled: boolean;
+}
+
+export default function NotificationSettings({
+  className,
+}: {
+  className: string;
+}) {
+  const { t } = useTranslation("auth");
+
+  const queryClient = useQueryClient();
+
+  const { data, error, isLoading } = useQuery<
+    GetNotificationSettingsRes.AsObject,
+    GrpcError
+  >(
+    notificationSettingsQueryKey,
+    service.notifications.getNotificationSettings
+  );
+
+  const mutation = useMutation<
+    SetNotificationSettingsRes.AsObject,
+    GrpcError,
+    NotificationSettingFormData
+  >(
+    ({ newNotificationsEnabled }) =>
+      service.notifications.setNotificationSettings(newNotificationsEnabled),
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries(notificationSettingsQueryKey);
+      },
+    }
+  );
+
+  const toggleNewNotifications = async () => {
+    mutation.mutate({
+      newNotificationsEnabled: !data?.newNotificationsEnabled,
+    });
+  };
+
+  return (
+    <div className={className}>
+      <Typography variant="h2">{t("notification_settings.title")}</Typography>
+      {(isLoading || !data) ? (
+        <CircularProgress />
+      ) : error ? (
+        <Alert severity="error">{error.message}</Alert>
+      ) : (
+        <>
+          <Typography variant="body1">
+            The new notification system is currently{" "}
+            <b>{data.newNotificationsEnabled ? "enabled" : "disabled"}</b> for
+            your account.
+          </Typography>
+          {mutation.error ? (
+            <Alert severity="error">{mutation.error.message}</Alert>
+          ) : (
+            <Typography variant="body1">
+              <Button
+                onClick={() => toggleNewNotifications()}
+                loading={mutation.isLoading}
+              >
+                {data.newNotificationsEnabled ? "Disable" : "Enable"} new
+                notification system.
```

Comment:
Translation keys? Do a key each for the whole sentence (ie. no need to interpolate)

Reply:
I was thinking that this is a temporary beta feature and won't need to be translated. (Once the new notification system is accepted, the language will change)

Reply:
ok